src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ProjectsController.kt

```diff
@@ -96,7 +96,7 @@ class ProjectsController(private val session: Session,
             @RequestParam("note") note: String): ResponseEntity<String>
     {
         projectRepository.updateProjectNote(projectId, userId(), note)
-        return EmptySuccessResponse.asResponseEntity()
+        return SuccessResponse(projectRepository.getProjects(userId())).asResponseEntity()
```

Comment:
What's the need for this change? Shouldn't need to get all projects having just updated a note field. 

Is it because you want to refresh all projects having updated the note? Our standard pattern for that is to refetch from the projects endpoint when the initial request succeeds, as [here](https://github.com/mrc-ide/hint/blob/master/src/app/static/src/app/store/projects/actions.ts#L190) and [here](https://github.com/mrc-ide/hint/blob/master/src/app/static/src/app/store/projects/actions.ts#L206) etc. It does mean making a second request, but it means each endpoint has one job to do rather than sharing functionality. 

This is the sort of change it would be great to explain your reasoning for in the PR.

Reply:
This changes updates the list of project without having to despatch a getProject. While the getProject will retrieve all projects but it causes a refresh of project history page, hence closes the collapsible. I do not think we have any efficiency issue and I didn't set loading flag for setting previous project because I do not want any wait before rendering. I also didn't return a project because the list uses date meta to rank the displayed projects. This works nicely, what's the issue ?

Reply:
> This works nicely, what's the issue ?

As I said, my issue is that it differs from the other endpoints which make changes to the projects so it breaks our general pattern where a second request is made to retrieve projects, and as @mwoodbri says, it also overloads the endpoint to do more than it should need to. 

You make a good point about the subsequent refresh of the projects page collapsing the open project but perhaps there's another way to approach that e.g. adding a parameter to `getProjects` to avoid doing `SetLoading` (it's that which causes the project on the page to collapse?)